src/tasks.py

```diff
@@ -290,3 +329,18 @@ def run_yara_batch(job_id: JobId, iterator: str, batch_size: int) -> None:
 
         agent.execute_yara(job, pop_result.files)
         agent.add_tasks_in_progress(job, -1)
+
+
+def read_bytes_with_context(
+    data: bytes, matched_length: int, offset: int, byte_range: int = 32
+) -> tuple[bytes, bytes, bytes]:
+    """Return `matched_length` bytes from `offset`, along with `byte_range` bytes before and after the match."""
+
+    before = data[max(0, offset - byte_range) : offset]
+    matching = data[offset : offset + matched_length]
+    after = data[
+        offset
+        + matched_length : min(len(data), offset + matched_length + byte_range)
+    ]
```

Comment:
nitpick, but we save 2 LoC - only negative numbers are a problem so this works too
```suggestion
    before = data[max(0, offset - byte_range) : offset]
    matching = data[offset : offset + matched_length]
    after = data[offset + matched_length : offset + matched_length + byte_range]
```